### PR TITLE
i18n entity get translations

### LINKED ISSUE #17: I18N - Check Entity Has Translation
Boolean to check if the given entity has the given translation

```
bamboo_i18n_entity_has_translation(entity, langcode)
```

### PR DESCRIPTION
**Pull request by WengerK**
Will close #17 & #19

### CODE REVIEW

File: bamboo_twig_i18n/src/TwigExtension/I18n.php [hunk 1]
```diff
@@ -85,4 +87,33 @@ public function formatDate(TwigEnvironment $env, $date, $type = 'medium', $forma
...
+    /** @var \Drupal\Core\Language\LanguageManagerInterface $language_manager */
+    $language_manager = $this->getLanguageManager();
+
+    if (!$langcode) {
+      $langcode = $this->getCurrentLanguage();
+    }
+
+    // Check the given langcode exists.
+    $languages = $language_manager->getLanguages();
+    if (!isset($languages[$langcode])) {
```
Comment (gido) [inline bamboo_twig_i18n/src/TwigExtension/I18n.php]: not necessary, `$entity->hasTranslation()` do this check.

File: bamboo_twig_i18n/src/TwigExtension/I18n.php [hunk 2]
```diff
@@ -85,4 +87,33 @@ public function formatDate(TwigEnvironment $env, $date, $type = 'medium', $forma
...
   }
 
+  /**
+   * Gets a translation of the entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity for which to get the translation.
+   * @param string|null $langcode
+   *   (optional) The language code of the translation to get. NULL (default) means to use
+   *   the user interface language for the page.
```
Comment (gido) [inline bamboo_twig_i18n/src/TwigExtension/I18n.php:97]: `user interface language` or `current language`. Are they the same thing?

File: bamboo_twig_i18n/src/TwigExtension/I18n.php [hunk 3]
```diff
@@ -85,4 +87,33 @@ public function formatDate(TwigEnvironment $env, $date, $type = 'medium', $forma
...
+    }
+
+    // Check the given langcode exists.
+    $languages = $language_manager->getLanguages();
+    if (!isset($languages[$langcode])) {
+      return NULL;
+    }
+
+    // Return the translation of this entity or NULL when not translated.
+    return $entity->hasTranslation($langcode) ? $entity->getTranslation($langcode) : NULL;
```
Comment (gido) [inline bamboo_twig_i18n/src/TwigExtension/I18n.php]: So we agree to return `NULL` if the translation didn't exist ?

```twig
{{ entity|bamboo_i18n_get_translation('fr').title.value }}
```
will raise an error if `fr` translation didn't exist. It's fine for me and certainly a better API.

@tonifisler OK with this ?

Comment (gido) [reply bamboo_twig_i18n/src/TwigExtension/I18n.php]: We need to return `NULL`  to cover #17

**Review by vincentmoulene — COMMENTED** [review]
Nice improvements! 
Just need to add the changes of @gido.

File: bamboo_twig_i18n/src/TwigExtension/I18n.php (hunk 2 above)
Comment (WengerK) [reply bamboo_twig_i18n/src/TwigExtension/I18n.php:97]: it's the same.

**Pull request merged**